packages/core/src/payment/payment-method-request-sender.ts

```diff
@@ -44,4 +48,81 @@ export default class PaymentMethodRequestSender {
             params,
         });
     }
+
+    /**
+     * GraphQL payment requests
+     */
+    loadPaymentWalletWithInitializationData(
+        methodId: string,
+        options: HeadlessPaymentRequestOptions,
+    ): Promise<Response<PaymentMethod>> {
+        const url = `/graphql`;
+
+        const entityId = this.getPaymentEntityId(methodId);
+
+        const graphQLQuery = `
+            query {
+                site {
+                    paymentWalletWithInitializationData(filter: {paymentWalletEntityId: "${entityId}"}) {
+                        clientToken
+                        initializationData
+                    }
+                }
+            }
+        `;
+
+        const requestOptions: HeadlessPaymentRequestOptions = {
+            headers: {
+                ...options.headers,
+                'Content-Type': 'application/json',
+            },
+            body: {
+                query: graphQLQuery,
+            },
+        };
+
+        return this._requestSender
+            .post<HeadlessPaymentMethodResponse>(url, requestOptions)
+            .then((response) => this.transformToPaymentMethodResponse(response, methodId));
+    }
+
+    private transformToPaymentMethodResponse(
+        response: Response<HeadlessPaymentMethodResponse>,
+        methodId: string,
+    ): Response<PaymentMethod> {
+        const {
+            body: {
+                data: {
+                    site: {
+                        paymentWalletWithInitializationData: { clientToken, initializationData },
+                    },
+                },
+            },
+        } = response;
+
+        return {
+            ...response,
+            body: {
+                initializationData: initializationData
+                    ? JSON.parse(atob(initializationData))
+                    : null,
+                clientToken,
+                id: methodId,
+                config: {},
+                method: '',
+                supportedCards: [],
+                type: 'PAYMENT_TYPE_API',
+            },
+        };
+    }
+
+    private getPaymentEntityId(methodId: string): HeadlessPaymentMethodType {
+        const entityId = HeadlessPaymentMethodConfig[methodId];
+
+        if (!entityId) {
+            throw new Error('Unable to get payment entity id.');
+        }
```

Comment:
What would happen after throwing this error?

Reply:
We are not be able to make a request in this case. It will be catch with `PaymentMethodActionCreator` when i add it by the next PR